lib/file-list.js

```diff
@@ -40,214 +36,206 @@ function byPath (a, b) {
-  if (!path || !pattern) return
-  if (!this.buckets.has(pattern.pattern)) return
-
-  return _.find(Array.from(this.buckets.get(pattern.pattern)), function (file) {
-    return file.originalPath === path
-  })
-}
+  // Private Interface
+  // -----------------
+
+  // Is the given path matched by any exclusion filter
+  //
+  // path - String
+  //
+  // Returns `undefined` if no match, otherwise the matching
+  // pattern.
+  _isExcluded (path) {
+    return _.find(this._excludes, (pattern) => mm(path, pattern))
+  }
 
-// Is the given path already in the files list.
-//
-// path - String
-//
-// Returns a boolean.
-List.prototype._exists = function (path) {
-  var self = this
-
-  var patterns = this._patterns.filter(function (pattern) {
-    return mm(path, pattern.pattern)
-  })
-
-  return !!_.find(patterns, function (pattern) {
-    return self._findFile(path, pattern)
-  })
-}
+  // Find the matching include pattern for the given path.
+  //
+  // path - String
+  //
+  // Returns the match or `undefined` if none found.
+  _isIncluded (path) {
+    return _.find(this._patterns, (pattern) => mm(path, pattern.pattern))
+  }
 
-// Check if we are currently refreshing
-List.prototype._isRefreshing = function () {
-  return this._refreshing.isPending()
-}
+  // Find the given path in the bucket corresponding
+  // to the given pattern.
+  //
+  // path    - String
+  // pattern - Object
+  //
+  // Returns a File or undefined
+  _findFile (path, pattern) {
+    if (!path || !pattern) return
+    if (!this.buckets.has(pattern.pattern)) return
+
+    return _.find(Array.from(this.buckets.get(pattern.pattern)), (file) => {
+      return file.originalPath === path
+    })
+  }
 
-// Do the actual work of refreshing
-List.prototype._refresh = function () {
-  var self = this
-  var buckets = this.buckets
-  var matchedFiles = new Set()
+  // Is the given path already in the files list.
+  //
+  // path - String
+  //
+  // Returns a boolean.
+  _exists (path) {
+    const patterns = this._patterns.filter((pattern) => mm(path, pattern.pattern))
 
-  var promise = Promise.map(this._patterns, function (patternObject) {
-    var pattern = patternObject.pattern
-    var type = patternObject.type
+    return !!_.find(patterns, (pattern) => this._findFile(path, pattern))
+  }
 
-    if (helper.isUrlAbsolute(pattern)) {
-      buckets.set(pattern, new Set([new Url(pattern, type)]))
-      return Promise.resolve()
-    }
+  // Check if we are currently refreshing
+  _isRefreshing () {
+    return this._refreshing.isPending()
+  }
 
-    var mg = new Glob(pathLib.normalize(pattern), GLOB_OPTS)
-    var files = mg.found
-    buckets.set(pattern, new Set())
+  // Do the actual work of refreshing
+  _refresh () {
+    const buckets = this.buckets
+    const matchedFiles = new Set()
 
-    if (_.isEmpty(files)) {
-      log.warn('Pattern "%s" does not match any file.', pattern)
-      return
-    }
+    let promise
```

Comment:
It seems that let/const use a bit different initialization logic, than var. I had to split this into declaration and initialization for tests to pass. They override scheduler for `Promise` to invoke methods synchronously and it throws `promise is not defined` after conversion.